src/lib.rs

```diff
@@ -216,24 +220,47 @@ pub mod pallet {
 	pub(super) type UserTipsTotal<T> =
 		StorageMap<_, Blake2_128Concat, AccountIdOf<T>, BalanceOf<T>, ValueQuery>;
 	// Oracle
+	/// Accumulated staking reward per staked token
+	#[pallet::storage]
+	#[pallet::getter(fn accumulated_reward_per_share)]
+	pub(super) type AccumulatedRewardPerShare<T> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+	/// Mapping of query identifiers to a report.
 	#[pallet::storage]
 	pub(super) type Reports<T> = StorageMap<_, Blake2_128Concat, QueryId, ReportOf<T>>;
```

Comment:
Unrelated to this PR, but is `Blake2_128Concat` correct for `QueryId`? Shouldn't it be keccak256, or am I reading this line incorrectly?

Reply:
https://github.com/tellor-io/tellor-pallet/issues/25 references this, whereby currently the hasher creates a `Blake2` hash of the `QueryId`, which in itself a `Keccak256` hash. This could be optimised by changing from the `Blake2_128Concat` hasher to the Identity hasher, which passes the Keccak256 hash through unchanged.

Reply:
Oh, gotcha. Thanks